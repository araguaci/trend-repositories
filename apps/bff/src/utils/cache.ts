const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
