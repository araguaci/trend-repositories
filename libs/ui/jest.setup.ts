import '@testing-library/jest-dom';
jest.mock('next/dist/client/router', () => require('next-router-mock'));
