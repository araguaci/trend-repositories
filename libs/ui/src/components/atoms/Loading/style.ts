import styled from 'styled-components';

export const StyledLoading = styled.div`
  display: flex;
  justify-content: center;
  padding: 1rem;
`;
