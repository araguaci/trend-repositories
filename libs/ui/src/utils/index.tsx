/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
