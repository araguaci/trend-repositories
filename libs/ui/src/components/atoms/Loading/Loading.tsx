import { StyledLoading } from './style';
import { Text } from '../Text';

const Loading = () => {
  return (
    <StyledLoading>
      <Text size="large">Loading...</Text>
    </StyledLoading>
  );
};

export default Loading;
