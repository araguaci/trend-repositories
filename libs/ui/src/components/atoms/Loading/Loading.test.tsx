import { renderWithProvider } from '@trend-repositories/ui/utils';
import Loading from './Loading';

describe('Loading Component', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithProvider(<Loading />);

    expect(asFragment()).toMatchSnapshot();
  });
});
