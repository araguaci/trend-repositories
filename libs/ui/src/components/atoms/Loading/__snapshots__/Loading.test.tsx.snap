// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="style__StyledLoading-sc-131w25h-0 hFAxFx"
  >
    <span
      class="style__StyledText-sc-1aawx67-0 iWiWeI"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
